server/src/main/java/org/opensearch/gateway/remote/RemoteGlobalMetadataManager.java

```diff
@@ -0,0 +1,218 @@
+    private final RemoteClusterStateBlobStore<Custom, RemoteCustomMetadata> customMetadataBlobStore;
+    private final Compressor compressor;
+    private final NamedXContentRegistry namedXContentRegistry;
+
+    RemoteGlobalMetadataManager(
+        ClusterSettings clusterSettings,
+        RemoteWritableEntityStore<Metadata, RemoteGlobalMetadata> globalMetadataBlobStore,
+        RemoteClusterStateBlobStore<CoordinationMetadata, RemoteCoordinationMetadata> coordinationMetadataBlobStore,
+        RemoteClusterStateBlobStore<Settings, RemotePersistentSettingsMetadata> persistentSettingsBlobStore,
+        RemoteClusterStateBlobStore<TemplatesMetadata, RemoteTemplatesMetadata> templatesMetadataBlobStore,
+        RemoteClusterStateBlobStore<Custom, RemoteCustomMetadata> customMetadataBlobStore,
+        Compressor compressor,
+        NamedXContentRegistry namedXContentRegistry
+    ) {
+        this.globalMetadataUploadTimeout = clusterSettings.get(GLOBAL_METADATA_UPLOAD_TIMEOUT_SETTING);
+        this.compressor = compressor;
+        this.globalMetadataBlobStore = globalMetadataBlobStore;
+        this.namedXContentRegistry = namedXContentRegistry;
+        this.coordinationMetadataBlobStore = coordinationMetadataBlobStore;
+        this.persistentSettingsBlobStore = persistentSettingsBlobStore;
+        this.templatesMetadataBlobStore = templatesMetadataBlobStore;
+        this.customMetadataBlobStore = customMetadataBlobStore;
+        clusterSettings.addSettingsUpdateConsumer(GLOBAL_METADATA_UPLOAD_TIMEOUT_SETTING, this::setGlobalMetadataUploadTimeout);
+    }
+
+    /**
+     * Allows async upload of Metadata components to remote
+     */
+    CheckedRunnable<IOException> getAsyncMetadataWriteAction(
+        AbstractRemoteWritableBlobEntity writeEntity,
+        RemoteWritableEntityStore remoteStore,
+        LatchedActionListener<ClusterMetadataManifest.UploadedMetadata> latchedActionListener
+    ) {
+        return () -> remoteStore.writeAsync(writeEntity, getActionListener(writeEntity, latchedActionListener));
+    }
+
+    private ActionListener<Void> getActionListener(
+        AbstractRemoteWritableBlobEntity remoteBlobStoreObject,
+        LatchedActionListener<ClusterMetadataManifest.UploadedMetadata> latchedActionListener
+    ) {
+        return ActionListener.wrap(
+            resp -> latchedActionListener.onResponse(remoteBlobStoreObject.getUploadedMetadata()),
+            ex -> latchedActionListener.onFailure(new RemoteStateTransferException("Upload failed", ex))
+        );
+    }
+
+    Metadata getGlobalMetadata(String clusterUUID, ClusterMetadataManifest clusterMetadataManifest) {
+        String globalMetadataFileName = clusterMetadataManifest.getGlobalMetadataFileName();
+        try {
+            // Fetch Global metadata
+            if (globalMetadataFileName != null) {
+                RemoteGlobalMetadata remoteGlobalMetadata = new RemoteGlobalMetadata(
+                    globalMetadataFileName,
```

Comment:
This fileName is obtained from the manifest and does not contain .dat extension. This extension was being previously added from ChecksumBlobStoreFormat. But now that the logic for download has been moved to RemoteGlobalMetadata, we need add `.dat` extension from here.

Reply:
Added a String format for this